### PR TITLE
[[ Docs ]] Update the LCB spec in the Extending LiveCode guide

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

File: Documentation/guides/Extending LiveCode.md [hunk 1]
```diff
@@ -1250,7 +1257,7 @@ Additionally, all types can be annotated with **optional**. An optional annotati
         : 'module' <Name: Identifier> SEPARATOR
             { Metadata SEPARATOR }
             { Import SEPARATOR }
-            { Definition SEPARATOR }
```
Comment (peter-b) [inline Documentation/guides/Extending LiveCode.md:1253]: There's no difference between `use` and any other definition now, so you can get rid of the separate `Metadata` and `Import` lines and just include them in the option clause.

File: Documentation/guides/Extending LiveCode.md [hunk 2]
```diff
@@ -1360,12 +1368,29 @@ The remaining types are as follows:
  - **nothing**: a single value *nothing* (this is used to describe handlers with no return value - i.e. void)
  - **Pointer**: a low-level pointer (this is used with foreign code interconnect and shouldn't be generally used).
 
-> **Note:** *Integer* and *Real* are currently the same as *Number*.
+> **Note:** *integer* and *real* are currently the same as *number*.
 
-> **Note:** In a subsequent update you will be able to specify lists and arrays of fixed types. For example, *List of String*.
+> **Note:** In a subsequent update you will be able to specify lists and arrays of fixed types. For example, *list of string*.
```
Comment (peter-b) [inline Documentation/guides/Extending LiveCode.md]: These changes are incorrect; even builtin type names (e.g. `Integer`, `List`, `String`) are now `TitleCase`.

**Comment by livecodeali** [comment]
Thanks, I took these changes from
https://github.com/livecode/livecode/blob/develop/docs/specs/livecode_builder_language_reference.md

I don't know if there's any way of not having to do this manually :-1:

**Comment by peter-b** [comment]
@livecode-vulcan review ok 8778a7c

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 8778a7c301437de83c0413ae208bd1d780f3504a

**Pull request merged**